src/LogicalOpFactory.php

```diff
@@ -92,13 +92,15 @@ protected function splitByLogicOp($expression)
             return [ [$expression, '', $expression] ];
         }
         $exprSet = [];
-        $i = 0;
-        foreach ($matches as $opWithOffset) {
-            list($op, $offset) = $opWithOffset[0];
```

Comment:
If there is any logic operator, there are always exactly 2 entries in `$matches`, so `$opWithOffset[0]` is always the same. We must iterate over `$matches[0]` (or `$matches[1]`) to find each individual expression.